### PR TITLE
Update Infineon XDPE programming procedure

### PR DESCRIPTION
**Pull request by aasboddu**
Update the programming procedure as per the
AN001-XDPE1x2xx (XDPE192xx RevD & XDPE1D2xx RevB Programming Guide)

### CODE REVIEW

File: src/vr_update_infineon_xdpe.cpp [hunk 1]
```diff
@@ -26,35 +26,39 @@ bool vr_update_infineon_xdpe::crcCheckSum()
...
 
     rc = i2c_smbus_write_byte_data(fd, BYTE_PREFIX, GET_CRC);
-
     if (rc != SUCCESS)
     {
-        sd_journal_print(LOG_ERR, "Error: Failed to write data\n");
+        sd_journal_print(LOG_ERR,
+        "Error: Failed to read CRC using "
+        "MFR Cmd: GET_CRC (0x%x), rc=%d\n",
+        GET_CRC);
```
Comment (abinayaddhandapani) [inline src/vr_update_infineon_xdpe.cpp]: rc variable is not specified for print.

File: src/vr_update_infineon_xdpe.cpp [hunk 2]
```diff
@@ -84,84 +90,100 @@ bool vr_update_infineon_xdpe::isUpdatable()
...
 
     rc = i2c_smbus_write_byte_data(fd, BYTE_PREFIX, AVAIL_SPACE_BYTE);
-
     if (rc != SUCCESS)
     {
-        sd_journal_print(LOG_ERR, "Error: Failed to write data\n");
+        sd_journal_print(LOG_ERR, 
+        "Error: Failed to read available space using "
+        "MFR Cmd: OTP_PARTITION_SIZE_REMAINING (0x%x), rc=%d\n", 
+        AVAIL_SPACE_BYTE);
```
Comment (abinayaddhandapani) [inline src/vr_update_infineon_xdpe.cpp]: rc is not specified in the print

File: src/vr_update_infineon_xdpe.cpp (hunk 1 above)
Comment (aasboddu) [reply src/vr_update_infineon_xdpe.cpp]: ack

File: src/vr_update_infineon_xdpe.cpp (hunk 2 above)
Comment (aasboddu) [reply src/vr_update_infineon_xdpe.cpp]: ack

**Review by abinayaddhandapani — APPROVED** [review]

File: src/vr_update_infineon_xdpe.cpp [hunk 3]
```diff
@@ -26,35 +26,39 @@ bool vr_update_infineon_xdpe::crcCheckSum()
...
+        "Error: Failed to read CRC using "
+        "MFR Cmd: GET_CRC (0x%x), rc=%d\n",
+        GET_CRC, rc);
         return false;
     }
 
-    /*Wait for 200ms */
-    usleep(200000);
+    /*Wait for 20ms */
+    usleep(20000);
```
Comment (supven01) [inline src/vr_update_infineon_xdpe.cpp]: magic number 20000

File: src/vr_update_infineon_xdpe.cpp [hunk 4]
```diff
@@ -422,7 +493,8 @@ bool vr_update_infineon_xdpe::UpdateFirmware()
 
     std::vector<std::vector<std::string>> sections = parseCfgFile(ConfigFilePath);
 
-    std::string trim_header = "00000002";    // Trim header programming needs to be ignored according to Infineon FAE
+    // Trim header programming needs to be ignored according to Infineon FAE
+    std::string trim_header = "00000002";
```
Comment (supven01) [inline src/vr_update_infineon_xdpe.cpp]: magic number

**Review by supven01 — CHANGES_REQUESTED** [review]

File: src/vr_update_infineon_xdpe.cpp [hunk 5]
```diff
@@ -331,21 +357,59 @@ int writeDataToScratchpad(std::vector<std::string> section,int fd)
 {
     uint8_t wdata[MAXBUFFERSIZE];
     int rc = FAILURE;
-    wdata[INDEX_0] = SPBD0;
-    wdata[INDEX_1] = SPBD1;
-    wdata[INDEX_2] = SPBD2;
-    wdata[INDEX_3] = SPBD3;
+    wdata[INDEX_0] = 0x2;
```
Comment (supven01) [inline src/vr_update_infineon_xdpe.cpp:360]: This got changed from preprocessor macro to magic numbers.
I can overlook this for now, but please fix this next time.

**Review by supven01 — APPROVED** [review]

**Pull request merged**